### DISCUSSION TITLE
Add feature: Reset OC settings after exiting miner

### DISCUSSION THREAD

**Discussion by PedroAlfNeto** [NebuTech/NBMiner#723] [Ideas]
The new overclock options via parameters are a trully awesome feature, especially for switching between profitable algorithms for lhr gpus that beneffit from different OC settings.

Proposed feature:
- A cool addition would be to reset the OC settings when closing nbminer, especially for gamers who loose performance with mining OC.

Not sure if it would be easy if the nbminer task is closed abruptly, leaving no time for nbminer to reset OC setings.

**Comment by Tom5051** [comment]
same with the GPU fan settings

**Comment by stewevans** [comment]
This would be so awesome. The overclock settings are amazing and work really well, but they are not usable for gamer/miners when there is no good way to reset.

Is there a fix for this comming, or a recommended way to reset?

I've tried running another nbminer start batch file with zeroed out OC settings which seems to work, but that's a bit clunky. It's also not exactly clear in the docs what 0 settings do? Restarting the computer also appears to reset.

Perhaps installing afterburner could do this by creating profiles, but that's kind of defeating the purpose for having the OC settings in nbminer in the first place. Would really just be great if it can be reset on close?

**Reply by LuxKeiwoker** [reply]
Once nbminer has set the settings, you can't even change the coreclock anymore in Afterburner. It's stuck to whatever coreclock you have set in the batch-file.

**Comment by aelberone** [comment]
This could be solved with a script like:
```
#!/bin/bash
nvidia-settings -a GPUMemoryTransferRateOffsetAllPerformanceLevels=-2000
sudo ./nbminer ...
# not executed because of CTRL+C
nvidia-settings -a GPUMemoryTransferRateOffsetAllPerformanceLevels=0
```

... but ... CTRL+C stop also script.
So, please add  '**q to Quit**'
